File: kt/Minsk/minsk/src/main/kotlin/dev/phytolizer/minsk/analysis/text/TextLine.kt
package dev.phytolizer.minsk.analysis.text

class TextLine(private val _text: SourceText, val start: Int, val length: Int, val lengthIncludingLineBreak: Int) {
    val end: Int
        get() = start + length
    val endIncludingLineBreak: Int
        get() = start + lengthIncludingLineBreak
    val span: TextSpan
        get() = TextSpan(start, length)
    val spanIncludingLineBreak: TextSpan
        get() = TextSpan(start, lengthIncludingLineBreak)
}

File: cs/Minsk/CodeAnalysis/Text/TextLine.cs
namespace Minsk.CodeAnalysis.Text;

public sealed record TextLine(SourceText Text, int Start, int Length, int LengthIncludingLineBreak)
{
    public int End => Start + Length;
    public TextSpan Span => new(Start, Length);
    public TextSpan SpanIncludingLineBreak => new(Start, LengthIncludingLineBreak);

    public override string ToString()
    {
        return Text.ToString(Span);
    }
}
